fontforge/parsettf.c

```diff
@@ -4241,71 +4241,73 @@ return( -1 );
     if ( modtype<=1 /* Unicode */ ) {
 	/* No conversion needed, already unicode */;
     } else if ( modtype==2 /* SJIS */ ) {
-	if ( enc<=127 ) {
-	    /* Latin */
-	    if ( enc=='\\' ) enc = 0xa5;	/* Yen */
-	} else if ( enc>=161 && enc<=223 ) {
-	    /* Katakana */
-	    enc = unicode_from_jis201[enc];
-	} else if ( enc<255 ) {
-	    /* This is erroneous as I understand SJIS */
-	    enc = badencoding(info);
-	} else if (enc >= 0xeaa5) {
-        /* Encoded value is outside SJIS range */
-        /* If this happens, it's likely that it's actually CP932 encoded */
-        /* Todo: Detect CP932 encoding earlier and apply that instead of SJIS */
-        enc = badencoding(info);
-	} else {
-	    int ch1 = enc>>8, ch2 = enc&0xff;
-	    if ( ch1 >= 129 && ch1<= 159 )
-		ch1 -= 112;
-	    else
-		ch1 -= 176;
-	    ch1 <<= 1;
-	    if ( ch2>=159 )
-		ch2-= 126;
-	    else if ( ch2>127 ) {
-		--ch1;
-		ch2 -= 32;
-	    } else {
-		--ch1;
-		ch2 -= 31;
-	    }
-	    if ( ch1<0x21 || ch2<0x21 || ch1>0x7e || ch2>0x7e )
-		enc = badencoding(info);
-	    else
-		enc = unicode_from_jis208[(ch1-0x21)*94+(ch2-0x21)];
-	}
-    } else if ( modtype==3 /* GB2312 offset by 0x8080, parse just like wansung */ ) {
-	if ( enc>0xa1a1 ) {
-	    enc -= 0xa1a1;
-	    enc = (enc>>8)*94 + (enc&0xff);
-	    enc = unicode_from_gb2312[enc];
-	    if ( enc==0 ) enc = -1;
-	} else if ( enc>0x100 )
-	    enc = badencoding(info);
+        // Apart from these, cp932 is a strict superset of sjis
+        if (enc == '\\')
+            enc = 0xa5; /* Yen */
+        else if (enc == 0x7e)
```

Comment:
See https://en.wikipedia.org/wiki/Code_page_932_(Microsoft_Windows)#Single-byte_character_differences for more info (+ #2820). From my sample size of 2, the Yen mapping is pretty common, and flip a coin for the tilde mapping.

At the least this keeps the same behaviour as before.

Reply:
Sorry, the cropping on this was weird I meant what's the point of:
```
                else if (enc == 0x7e)
                     enc = 0x7e; /* Tilde */
```

Reply:
Ah right, remnants of when I was using (iconv) SJIS instead of CP932. 

```
jeremy@ubuntu:~$ iconv -f CP932 -t UTF-8 test.txt
\~
jeremy@ubuntu:~$ iconv -f SJIS -t UTF-8 test.txt
¥‾
```

Guess it's more or less redundant now.

Reply:
Better to eliminate the guard if it's not needed or make the next one into `if (enc != 0x7e)`